KsApi/MockService.swift

```diff
@@ -300,6 +302,8 @@
 
       self.addNewCreditCardResult = addNewCreditCardResult
 
+      self.apolloClient = MockGraphQLClient.shared.client
```

Comment:
So agree that it should be passed in.
Strange issue occuring when initializing an `ApolloClient` instance in the `Service` intializer ala:
`apolloClient: ApolloClientType = ApolloClient(url: URL(string: "https://kickstarter.com")!)`
It produces this error when running Library/Kickstarter-Framework tests:
```
2021-08-20 14:24:16.861428-0400 xctest[93884:634852] The bundle “Library-iOSTests” couldn’t be loaded because it is damaged or missing necessary resources. Try reinstalling the bundle.
2021-08-20 14:24:16.861639-0400 xctest[93884:634852] (dlopen_preflight(/Users/mubaraksadoon/Library/Developer/Xcode/DerivedData/Kickstarter-fottjzxsklpdhffieterqczvcgbv/Build/Products/Debug-iphonesimulator/Library-iOSTests.xctest/Library-iOSTests): Library not loaded: @rpath/Apollo.framework/Apollo
  Referenced from: /Users/mubaraksadoon/Library/Developer/Xcode/DerivedData/Kickstarter-fottjzxsklpdhffieterqczvcgbv/Build/Products/Debug-iphonesimulator/Library-iOSTests.xctest/Library-iOSTests
  Reason: image not found)
Program ended with exit code: 82
```

Anyway, was able to get around this by not intializing the `ApolloClient` above in the init func params, but allowing it to be `nil` by default and initializing it in the function itself as `ApolloClient(url: URL(string: "https://kickstarter.com")!)`

it's dummy anyway, so it's never used.

Reply:
and it wasn't related to the import missing above.

Reply:
it's nil now

Reply:
That's strange, did you try the good old deleting derived data, clean and rebuild? Doesn't seem like something application level code would cause.

Reply:
I did indeed, still happened. Closed xcode, cleaned derived data, clean project, built and ran with a reset sim.

Reply:
Ok I think your current approach should be fine. We can continue to monitor this closely.